OmniKit/MessageTransport/MessageBlocks/StatusError.swift

```diff
@@ -9,16 +9,137 @@
+    public let secondaryLoggedFaultEvent: UInt8
+    public let logEventError: Bool
+    public let infoLoggedFaultEvent: InfoLoggedFaultEventType
+    public let progressAtFirstLoggedFaultEvent: ProgressType
+    public let recieverLowGain: UInt8
+    public let radioRSSI: UInt8
+    public let progressAtFirstLoggedFaultEventCheck: ProgressType
 
+    public let data: Data
+    
     public init(encodedData: Data) throws {
-        self.data = encodedData
+        
+        if encodedData.count < Int(13) {
+            throw MessageBlockError.notEnoughData
+        }
+        
+        self.length = encodedData[1]
+ 
+        guard let requestedType = GetStatusCommand.StatusType(rawValue: encodedData[2]) else {
+            throw MessageError.unknownValue(value: encodedData[2], typeDescription: "StatusType")
+        }
+        self.requestedType = requestedType
+
+        guard let progressType = ProgressType(rawValue: encodedData[3]) else {
+            throw MessageError.unknownValue(value: encodedData[3], typeDescription: "ProgressType")
+        }
+        self.progressType = progressType
+
+        guard let deliveryInProgressType = DeliveryInProgressType(rawValue: encodedData[4]) else {
+            throw MessageError.unknownValue(value: encodedData[4], typeDescription: "DeliveryInProgressType")
+        }
+        self.deliveryInProgressType = deliveryInProgressType
+
+        self.insulinNotDelivered = podPulseSize * Double((Int(encodedData[5] & 0x3) << 8) | Int(encodedData[6]))
+        
+        self.podMessageCounter = encodedData[7]
+        self.unknownPageCode = Double(Int(encodedData[8]) | Int(encodedData[9]))
+        
+        self.origionalLoggedFaultEvent = encodedData[10]
+        
+        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double((Int(encodedData[11] & 0b1) << 8) + Int(encodedData[12])))
+        
+        self.insulinRemaining = podPulseSize * Double((Int(encodedData[13] & 0x3) << 8) | Int(encodedData[14]))
+        
+        self.timeActive = TimeInterval(minutes: Double((Int(encodedData[15] & 0b1) << 8) + Int(encodedData[16])))
+        
+        self.secondaryLoggedFaultEvent = encodedData[17]
+
+        self.logEventError = encodedData[18] == 2
+
+        guard let infoLoggedFaultEventType = InfoLoggedFaultEventType(rawValue: encodedData[19] >> 4) else {
+            throw MessageError.unknownValue(value: encodedData[19] >> 4, typeDescription: "InfoLoggedFaultEventType")
+        }
+        self.infoLoggedFaultEvent = infoLoggedFaultEventType
+        
+        guard let progressAtFirstLoggedFaultEventType = ProgressType(rawValue: encodedData[19] & 0xF) else {
+            throw MessageError.unknownValue(value: encodedData[19] & 0xF, typeDescription: "ProgressType")
+        }
+        self.progressAtFirstLoggedFaultEvent = progressAtFirstLoggedFaultEventType
+        
+        self.recieverLowGain = encodedData[20] >> 4
+        
+        self.radioRSSI =  encodedData[20] & 0xF
```

Comment:
Is this a db field? Perhaps it can be converted to the actual value, if we know how to do it; it'd be interesting to see how strongly the pod hears us.